Implement a basic `Severing Fang` mob TP skill

mrhappyasthma opened the pull request:
This implementation is largely based off of `razor_fang.lua`.

The skill is set to not trigger for targets behind the mob, since sub-zero smash should always be triggered in that case.

I also adjusted the shape and range in the .sql file. The range value was chosen based off of the one from `razor_fang`. And the shape was updated to be a cone since the shape is a "front arc".

<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

mrhappyasthma committed:
Implement a basic 'Severing Fang' mob TP skill

This implementation is largely based off of `razor_fang.lua`.

The skill is set to not trigger for targets behind the mob, since sub-zero smash should always be triggered in that case.

I also adjusted the shape and range in the .sql file. The range value was chosen based off of the one from `razor_fang`. And the shape was updated to be a cone since the shape is a "front arc".

claywar commented on scripts/globals/mobskills/severing_fang.lua:
```diff
@@ -0,0 +1,38 @@
+  -- should trigger in this case.
+  if target:isBehind(mob) then
+    return 1
+  end
```
Work looks really reasonable across all PRs, the only thing I'd suggest is a bit of styling changes.  4-space indents as a standard is what we're holding to right now. 👍

mrhappyasthma replied:
Ah, force of habit from work (we use 2 spaces).

Please nitpick any inconsistencies because I may miss a few...

Will fix.

claywar commented on scripts/globals/mobskills/severing_fang.lua:
```diff
@@ -0,0 +1,38 @@
+    local dmg = MobFinalAdjustments(info.dmg, mob, skill, target, xi.attackType.PHYSICAL, xi.damageType.PIERCING, MOBPARAM_WIPE_SHADOWS)
+
+    local typeEffect = xi.effect.DEFENSE_DOWN
+    MobPhysicalStatusEffectMove(mob, target, skill, typeEffect, 30, 0, 60)
```
Minor at best; is there a reason the last local variable is broken out from the grouping above?  Again, nothing major, just curious.

mrhappyasthma replied:
I'm happy to do whatever the project prefers.

I'm a strong believer in placing local variables as close to their usages as possible. I believe it helps with readability (you don't have to jump earlier in the file to find out what value it holds) and it narrows the scope (i.e. lifetime) of the local.

But if the coding convention is to declare all locals ahead of the primary logic, I'm happy to move it up. Whatever is most consistent with the project.

claywar replied:
Honestly? If it makes sense, I'm on board with it.  That's a huge block of variables above, and I can see the reasoning here.

mrhappyasthma replied:
I'll leave this open for now. I'm interested to see other folk's opinions.

(This may be another instance of my work-style-guide heavily influencing my coding style haha.)

TeoTwawki replied:
if its used multiple places in the script and or avoid a wide line by making it a var, or if the variable ever gets modified later in the code: its good its a variable.

In cases where "typeEffect" only is ever called once, the line is not too wide with it in there, and its never modified: my preference is not to put it into a variable in that case.

just my 2 cents.

mrhappyasthma replied:
Sounds good. I'll inline it then, in this case. I'm not sure (by that description) that the variable buys much: the line length isn't an issue, and the variable name adds no additional clarity.

TeoTwawki replied:
As far as _placement_ of _where_ to put a variable.. I only really care if the placement changes the scope. Sensible scope, and I am in the habit of putting all vars that have the same scope together, There's no rule on that I am away of /shrug so w/e works bro :+1:

mrhappyasthma committed:
Fix indentation: use 4 spaces instead of 2

Was my bad on that function. So hard to fight against muscle-memory :P

mrhappyasthma committed:
Inline the 'typeEffect' local.

Per the discussion, it removes an unnecessary local and doesn't significantly increase line lengths.

mrhappyasthma commented:
Part of https://github.com/LandSandBoat/server/issues/201

zach2good commented on scripts/globals/mobskills/severing_fang.lua:
```diff
@@ -0,0 +1,37 @@
+    local accmod = 1
+    local dmgmod = 1.5
+    local info = MobPhysicalMove(mob, target, skill, numhits, accmod, dmgmod, TP_NO_EFFECT)
+    local dmg = MobFinalAdjustments(info.dmg, mob, skill, target, xi.attackType.PHYSICAL, xi.damageType.PIERCING, MOBPARAM_WIPE_SHADOWS)
```
Header says: `--  Utsusemi/Blink absorb: 2 shadows`

claywar replied:
Looks like that aligns with the data recorded for the ability (in that when hit with it, it'll absorb two shadows)

mrhappyasthma replied:
The comment is correct. This was a copy/paste error from the script I branched off of.

Good eye!

It should be `info.hitslanded` instead of `MOBPARAM_WIPE_SHADOWS`.

mrhappyasthma replied:
Corrected in the latest commit.

mrhappyasthma committed:
Fix a copy/paste bug: only take shadows corresponding to each hit.

This was a copy paste error. It should not remove all shadows, but rather remove at most 2 based on the  `info.hitslanded`.
